docs/tls/CONFIGURATION.md

```diff
@@ -0,0 +1,164 @@
+# Configuring NLK to use TLS / mTLS guide
```

Comment:
TLS / mTLS between what?
Client side?  
backend side between the NLK tier and ingress controller?
What is the use case here?

Reply:
Assume the user is not an NGINX or K8s expert, explain all the things.

Reply:
Added a better description of how TLS / mTLS applies and the actors involved. The use case is described in the [Considerations](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/blob/use-tls-for-nginx-connection/docs/tls/README.md#considerations) section of the README.

Added a stronger description to both this document and the README.

Reply:
My only comment to that is this: due to search, individual docs need to stand on their own, even if they are pieces of a larger thing.  That is how I am reviewing these files.